rtic-monotonics/src/imxrt.rs

```diff
@@ -1,112 +1,163 @@
+pub mod prelude {
+    #[cfg(feature = "imxrt_gpt1")]
+    pub use crate::imxrt_gpt1_monotonic;
+    #[cfg(feature = "imxrt_gpt2")]
+    pub use crate::imxrt_gpt2_monotonic;
+
+    pub use crate::Monotonic;
+    pub use fugit::{self, ExtU64, ExtU64Ceil};
+}
 
 #[doc(hidden)]
 #[macro_export]
 macro_rules! __internal_create_imxrt_timer_interrupt {
-    ($mono_timer:ident, $timer:ident, $timer_token:ident) => {{
+    ($mono_backend:ident, $timer:ident) => {
         #[no_mangle]
         #[allow(non_snake_case)]
         unsafe extern "C" fn $timer() {
-            $crate::imxrt::$mono_timer::__tq().on_monotonic_interrupt();
+            use $crate::TimerQueueBackend;
+            $crate::imxrt::$mono_backend::timer_queue().on_monotonic_interrupt();
         }
+    };
+}
 
-        pub struct $timer_token;
+#[doc(hidden)]
+#[macro_export]
+macro_rules! __internal_create_imxrt_timer_struct {
+    ($name:ident, $mono_backend:ident, $timer:ident, $tick_rate_hz:expr) => {
+        /// A `Monotonic` based on the GPT peripheral.
+        struct $name;
 
-        unsafe impl $crate::InterruptToken<$crate::imxrt::$mono_timer> for $timer_token {}
+        impl $name {
+            /// Starts the `Monotonic`.
+            ///
+            /// This method must be called only once.
+            pub fn start(gpt: $crate::imxrt::ral::gpt::$timer) {
+                $crate::imxrt::$mono_backend::_start(gpt);
+            }
+        }
+
+        impl $crate::TimerQueueBasedMonotonic for $name {
+            type Backend = $crate::imxrt::$mono_backend;
+            type Instant = $crate::fugit::Instant<
+                <Self::Backend as $crate::TimerQueueBackend>::Ticks,
+                1,
+                { $tick_rate_hz },
+            >;
+            type Duration = $crate::fugit::Duration<
+                <Self::Backend as $crate::TimerQueueBackend>::Ticks,
+                1,
+                { $tick_rate_hz },
+            >;
+        }
 
-        $timer_token
-    }};
+        $crate::rtic_time::impl_embedded_hal_delay_fugit!($name);
+        #[cfg(feature = "embedded-hal-async")]
+        $crate::rtic_time::impl_embedded_hal_async_delay_fugit!($name);
+    };
 }
 
-/// Register the GPT1 interrupt for the monotonic.
+/// Create a GPT1 based monotonic and register the GPT1 interrupt for it.
+///
+/// See [`crate::imxrt`] for more details.
+///
+/// # Arguments
+///
+/// * `name` - The name that the monotonic type will have.
+/// * `tick_rate_hz` - The tick rate of the timer peripheral. It's the user's responsibility
+///                    to configure the peripheral to the given frequency before starting the
+///                    monotonic.
 #[cfg(feature = "imxrt_gpt1")]
 #[macro_export]
-macro_rules! create_imxrt_gpt1_token {
-    () => {{
-        $crate::__internal_create_imxrt_timer_interrupt!(Gpt1, GPT1, Gpt1Token)
-    }};
+macro_rules! imxrt_gpt1_monotonic {
+    ($name:ident, $tick_rate_hz:expr) => {
+        mod _interrupts {
```

Comment:
This mod doesn't have to be exposed; instead, define the function inside of the timer struct. Same for all other monotonics.